### PR TITLE
Updating variable %BRANCH% to repalce hardcoded value main

### PR DESCRIPTION
**Pull request by gizas**
- Bug

## What does this PR do?

Fixes the variable that produces the BRANCH value of daemonsets. The value was hardcoded. 

## Why is it important?

Relates to https://github.com/elastic/elastic-agent/pull/3246/files

## Checklist

- [x] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `./changelog/fragments` using the [changelog tool](https://github.com/elastic/elastic-agent#changelog)
- [ ] I have added an integration test or an E2E test

## How to test this PR locally

```bash
cd elastic-agent/deploy/kubernetes

make generate-k8s
```
Test file `elastic-agent/deploy/kubernetes/elastic-agent-stanadlone/elastic-agent-standalone.yaml`
that contains
```yaml
      #        wget -O - https://github.com/elastic/elastic-agent/archive/main.tar.gz | tar xz -C /etc/elastic-agent/inputs.d --strip=5 "elastic-agent-main/deploy/kubernetes/elastic-agent-standalone/templates.d"

```

```bash
cd elastic-agent/deploy/kubernetes

GENERATEKUSTOMIZE=tue make ci-create-kustomize
```
Test file `elastic-agent/deploy/kubernetes/elastic-agent-kustomize/default/elastic-agent-standalone/base/elastic-agent-standalone-daemonset.yaml`

that contains
```yaml
      #        wget -O - https://github.com/elastic/elastic-agent/archive/main.tar.gz | tar xz -C /etc/elastic-agent/inputs.d --strip=5 "elastic-agent-main/deploy/kubernetes/elastic-agent-standalone/templates.d"

```

## Related issues

- 

## Questions to ask yourself

- How are we going to support this in production? 
- How are we going to measure its adoption? 
- How are we going to debug this?
- What are the metrics I should take care of?
- ...

### CODE REVIEW

**Review by pierrehilbert — APPROVED** [review]
Thanks for the fix

File: deploy/kubernetes/Makefile [hunk 1]
```diff
@@ -115,6 +115,6 @@ $(ALL):
 	sed -e "s/%VERSION%/${BEAT_VERSION}/g" -e "s/%BRANCH%/${BRANCH_VERSION}/g" -e "s/hostNetwork: true/hostNetwork: false/g" -e "s/DaemonSet/StatefulSet/g" -e "s/agent-node-datastreams/agent-ksm-datastreams/g" -e "/name: elastic-agent-state/,+1 s/^/#/" -e "/path: \/var\/lib\/$@\/kube-system\/state/,+1 s/^/#/" $@/$@-daemonset.yaml > $(KUSTOMIZE_KSM_AUTOSHARDING)/$@/extra/$@-statefulset.yaml
 
 else
-	echo "No KSM templates generated. Please run: KUSTOMIZE=tue make ci-create-kustomize "
+	echo "No KSM templates generated. Please run: GENERATEKUSTOMIZE=tue make ci-create-kustomize "
```
Comment (constanca-m) [inline deploy/kubernetes/Makefile:118]: `true` instead of `tue`?

Comment (ChrsMark) [inline deploy/kubernetes/Makefile:118]:
```suggestion
	echo "No KSM templates generated. Please run: GENERATEKUSTOMIZE=true make ci-create-kustomize "
```

**Review by constanca-m — APPROVED** [review]

**Comment by pierrehilbert** [comment]
Unrelated known failure, I'm forcing the merge.

**Pull request merged**